peppy/project.py

```diff
@@ -441,20 +441,67 @@ def templates_folder(self):
         return os.path.join(os.path.dirname(__file__), "submit_templates")
 
 
-    @staticmethod
-    def infer_name(path_config_file):
+    def infer_name(self):
         """
-        Infer project name based on location of configuration file.
+        Infer project name from config file path.
+        
+        First assume the name is the folder in which the config file resides,
+        unless that folder is named "metadata", in which case the project name
+        is the parent of that folder.
+        
+        :param str path_config_file: path to the project's config file.
+        :return str: inferred name for project.
+        """
+        import os
+
+        if hasattr(self, "name"):
+            return(self.name)
+        
+        config_folder = os.path.dirname(self.config_file)
+        project_name = os.path.basename(config_folder)
+        
+        if project_name == "metadata":
+            project_name = os.path.basename(os.path.dirname(config_folder))
+
+        self.name = project_name
```

Comment:
Actually, why is the effect moving into this function rather living out in the constructor?

Reply:
it isn't moving... it was already in the function...

Reply:
It was in the constructor, but `infer_name` itself didn't set anything, at least not in the current `master` version, right? It looked like the value from `infer_name` was used by the constructor to set `name`

Reply:
Ah, now I see what you mean. Ok, I can go with that way. 

I think I did it this way because to me, the function name `infer_name` for some reason implied a side-effect. that felt more logical. but to you it doesn't -- so I will move the setting logic back into the constructor.

Reply:
Ha actually, I can see the side-effecting implication of `infer_name`. It would seem odd to call this from outside the class, though, both semantically and conceptually if it's already in the constructor. So either way--in the constructor or as a separate, non-static function as it is now--seems fine, but if it's as a separate function and the name stays the same, can it be made `_infer_name`?

Reply:
the only time I could see wanting to call this externally would be if you update something (like move it) and want to re-infer the name. Or, if you overrode this function you might want to call it. hence I have a slight preference for keeping it as `infer_name`, but if you think `_infer_name` is better I will change it. I think there is a chance someone would call it externally in some situation.

Reply:
Ah, no I'd missed that the attribute had been entirely removed from this function--feels even better now! Again, GH graphics team should provide the fuzzy blanket icon. I can't be the only one who'd use it.